apps/ideas/src/components/Card/IdeaCardList.js

```diff
@@ -1,71 +1,102 @@
-import React from 'react'
-import CircularIndeterminateLoader from '../Loader/CircularIndeterminateLoader'
-import IdeaCard from './IdeaCard'
-import StyledIdeaCards from './StyledIdeaCards/StyledIdeaCards'
+import React from 'react';
+import CircularIndeterminateLoader from '../Loader/CircularIndeterminateLoader';
+import IdeaCard from './IdeaCard';
+import StyledIdeaCards from './StyledIdeaCards/StyledIdeaCards';
+import { TextCardTitleBox } from './StyledIdeaCard';
 import StyledCardsWrapper from '../CardsWrapper/StyledCardsWrapper';
-import axios from "axios";
-import { env } from "../../utils/EnvironmentVariables";
+import axios from 'axios';
+import { env } from '../../utils/EnvironmentVariables';
+import ProjectFilter from './ProjectFilter/ProjectFilter';
 import Grid from '@mui/material/Grid';
 
-
-
 function IdeaCardList() {
   const [cards, setCards] = React.useState([]);
   const [selectedCard, setSelectedCard] = React.useState({});
   const [loading, setLoading] = React.useState(true);
-
+  const [sortedCards, setSortedCards] = React.useState([]);
+
+  const sortingConfigs = [
+    { value: 'default', label: '' },
+    {
+      value: 'descending:mostRecentCommentTime',
+      label: 'Recent Activity',
+    },
+    {
+      value: 'descending:updated_at',
+      label: 'Recent Ideas',
+    },
+    {
+      value: 'ascending:hourCommitmentMin',
+      label: 'Time Commitment',
+    },
+  ];
+
```

Comment:
re: sortingConfig structure

Generally, this is a good workable structure, though some improvements could be made to add "safety". With the current implementation, the strings "descending" and "ascending" denote the desired order in which cards will be sorted. The pros of this solution are that it makes working with it pretty fluid, and feels reminiscent of a SQL query. I like that aspect of it. On the flip side, that fluidity leaves it open to some potential confusion for future devs if there are any small spelling mistakes or syntax errors. The danger here is that, in cases like that, there won't be any trace of where the errors are arising if future devs use this component and things don't work how they expect. My recommendation would be to create something to better represent the fact that there is a discrete classification happening with `ascending` vs `descending` so that more verbose errors will be shown when things go wrong.

Some potential solutions I can think of right now include using enum values, using booleans, or creating a struct/class with appropriate attributes to use in place of each array element in `sortingConfigs`. Alternatively, on a __completely__ different path, the same safety could be ensured by keeping things as they are but adding some very explicit error handling inside of the component around this particular string.

Reply:
Good point Kris. I added the isAscending property back in as part of the sortingConfigs in the IdeaCardList component, and then I am checking to see whether it's true in the SortableDropdown component. Then I am handling the string manipulation on the SortableDropdown component by concatenating the "ascending:" or "descending:" string with the value that is passed in for each option. Now we only need to check the boolean for isAscending to handle this rather than rely on a dev to use the exact string we're looking for in the sortElements function. I updated the comments in the SortableDropdown component to provide some clarity.